src/DotNetWorker.Core/Context/Features/DefaultModelBindingFeature.cs

```diff
@@ -46,24 +46,35 @@ public DefaultModelBindingFeature(IEnumerable<IConverter> converters)
                     functionBindings.TriggerMetadata.TryGetValue(param.Name, out source);
                 }
 
-                var converterContext = new DefaultConverterContext(param, source, context);
+                var converterContext = new DefaultConverterContext(param.Type, source, context);
+
+                if (param.Properties != null)
+                {
+                    // Pass info about specific input converter type defined for this parameter if present.
+                    if (param.Properties.TryGetValue(PropertyBagKeys.ConverterType, out var converterTypeAssemblyFullName))
+                    {
+                        converterContext.Properties = new Dictionary<string, object>()
```

Comment:
I had the `Properties` as nullable type in the `ConverterContext` definition.
Now I made it non-nullable property and `DefaultConverterContext` is immutable now. A constructor overload which takes the property bag also been added.  

But, this class still has to create a new dictionary instance if there is additional props to be passed. Is there a better way to handle it ? Currently the properties type is `IReadOnlyDictionary` in `ConverterContext`

it so that it is not nullable and the . Latest iteration should reflect those changes.